server/service/src/invoice_line/outbound_shipment_line/update/mod.rs

```diff
@@ -128,3 +128,243 @@ where
+                    r.id = mock_outbound_shipment_a_invoice_lines()[0].id.clone();
+                    r.stock_line_id = Some(mock_stock_line_on_hold()[0].id.clone());
+                    r.item_id = Some(mock_stock_line_on_hold()[0].item_id.clone());
+                }),
+            ),
+            Err(ServiceError::BatchIsOnHold)
+        );
+
+        // ReductionBelowZero
+        assert_eq!(
+            service.update_outbound_shipment_line(
+                &context,
+                &mock_store_a().id,
+                inline_init(|r: &mut UpdateOutboundShipmentLine| {
+                    r.id = mock_outbound_shipment_a_invoice_lines()[0].id.clone();
+                    r.number_of_packs = Some(100);
+                }),
+            ),
+            Err(ServiceError::ReductionBelowZero {
+                stock_line_id: mock_outbound_shipment_a_invoice_lines()[0]
+                    .stock_line_id
+                    .clone()
+                    .unwrap(),
+                line_id: mock_outbound_shipment_a_invoice_lines()[0].id.clone(),
+            })
+        );
+
+        // StockLineAlreadyExistsInInvoice
+        assert_eq!(
+            service.update_outbound_shipment_line(
+                &context,
+                &mock_store_a().id,
+                inline_init(|r: &mut UpdateOutboundShipmentLine| {
+                    r.id = mock_outbound_shipment_a_invoice_lines()[0].id.clone();
+                    r.stock_line_id = Some(mock_item_b_lines()[0].id.clone());
+                }),
+            ),
+            Err(ServiceError::StockLineAlreadyExistsInInvoice(
+                mock_outbound_shipment_a_invoice_lines()[1].id.clone()
+            ))
+        );
+
+        //TODO DatabaseError, InvoiceDoesNotExist, NotThisStoreInvoice
+    }
+
+    #[actix_rt::test]
+    async fn update_outbound_shipment_line_success() {
+        let (_, connection, connection_manager, _) = setup_all(
+            "update_outbound_shipment_line_success",
+            MockDataInserts::all(),
+        )
+        .await;
+
+        let service_provider = ServiceProvider::new(connection_manager, "app_data");
+        let context = service_provider.context().unwrap();
+        let service = service_provider.invoice_line_service;
+
+        service
+            .update_outbound_shipment_line(
+                &context,
+                &mock_store_a().id,
+                inline_init(|r: &mut UpdateOutboundShipmentLine| {
+                    r.id = mock_outbound_shipment_c_invoice_lines()[0].id.clone();
+                    r.number_of_packs = Some(3);
+                    r.total_before_tax = Some(10.99);
+                }),
+            )
+            .unwrap();
+
+        let outbound_line = InvoiceLineRowRepository::new(&connection)
+            .find_one_by_id(&mock_outbound_shipment_c_invoice_lines()[0].id.clone())
+            .unwrap();
+
+        assert_eq!(
+            outbound_line,
+            inline_edit(&outbound_line, |mut u| {
```

Comment:
I think it's better to inline_edit on mock_outbound_shipment_c_invoice_lines()[0].

Also similar test for stock line adjustments as before (or similar logic to test please)